Revamp userguide

Peter9192 opened the pull request:
In addressing #365 and #341 I felt the need to restructure the user guide, so here we go. 

Docs preview: https://wam2layers--367.org.readthedocs.build/en/367/

Peter9192 commented:
This is ready to be merged together with #368, however we still need to add the config and region files to the 4TU upload in order for it to work.

BSchilperoort commented on docs/userguide/preprocess.md:
```diff
@@ -0,0 +1,79 @@
+## Data checking utilities
+
+To aid in pre-processing, WAM2layers comes with some basic data checking
+utilities. Specifically, the function `check_input` takes a single input file
```
The function is in `wam2layers.analysis.checks`, and currently has no docstring. If you refer to it from the documentation it might be better to at least add a docstring, and have it included in the documentation (like an API docs page).

Peter9192 replied:
Good point. I've opened #384 and temporarily removed this paragraph from the docs. It's useful to have it, but currently I think it's outdated. Perhaps you could pick this up while reworking the input for pressure levels and adding cmip?

BSchilperoort reviewed (changes requested):
I was able to download the cases and run them after changing into the new directories.

The docs look good, I did leave some comments.

There currently is no documentation on how to maintain the example cases. If a developer wants to add or modify an example case, it would be nice to have a simple overview on what they would need to do and how to structure the files.

Peter9192 commented on docs/userguide/installation.md:
```diff
@@ -0,0 +1,120 @@
+
+To run WAM2Layers you need a computer and some kind of command line, such as the
+default terminal included in Mac or Linux. On Windows we recommend using Ubuntu
+through [WSL](https://learn.microsoft.com/en-us/windows/wsl/install). If you're
```
Do you know of any good pointer for getting started with python on windows? Most I've seen is people setting up Spyder or Anaconda prompt. VSCode with integrated terminal could be fine, but default cmd is very limited and no-one knows how to use powershell. Is git bash the best alternative then?

BSchilperoort replied:
Many basic powershell commands are the same as the ones on e.g. linux nowadays (cd, ls, ...), so using it at a basic level is quite easy.

wam2layer user setup on Windows:
- Install Python from python.org
- Open Powershell (cmd is also possible, see [this](https://docs.python.org/3/library/venv.html))
  - Create python virtual environment (`python -m venv env_name`)
  - Activate environment `env_name\Scripts\Activate.ps1`
  - `pip install wam2layers`
  - Run wam2layers in the terminal like normal

Or if don't bother with virtual environments, just install python, open the python shell which will be in the startup menu, and `pip install wam2layers`.

Peter9192 replied:
I've opened #383 to add this info. The one thing I'm missing is how to add python to PATH on windows.

BSchilperoort replied:
> The one thing I'm missing is how to add python to PATH on windows.

That just an option in the installer.

Peter9192 commented:
Thanks for the review! I've addressed suggestions wrt the new text. Other good ideas on existing text (though moved and thus showing up in the diffs) can be addressed in separate PRs, and therefore I've opened some new issues. Can you approve and merge now?
